### ISSUE TITLE
How do I set parameters to automatically end training after the loss is stable?

### ISSUE THREAD

**Issue by DonnieZhang586** [pytorch/fairseq#1580]
environment：python3.6,torch1.2.0,gcc 4.9 cuda 10  Graphics card：titan 
train code:
CUDA_VISIBLE_DEVICES=2,3,6 python train.py data-bin/data --lr 0.5 --ddp-backend no_c10d --clip-norm 0.1 --min-lr 1e-09 --dropout 0.2 --optimizer sgd --batch-size 128 --max-tokens 4096 --max-epoch 500 --criterion cross_entropy --no-epoch-checkpoints  --arch fconv --save-dir checkpoints/fconv15

1、How do I set parameters to automatically end training after the loss is stable?
2、May I use --fp 16?，When I use the parameter --fp16, it appears：
data-bin/DBNQA_data train en-sparql 715600 examples
| epoch 001:   0%| | 9/1864 [00:06<21:09,  1.46it/s, loss=16.447, ppl=| WARNING: overflow detected, setting loss scale to: 64.0
| epoch 001:   1%| | 17/1864 [00:12<18:18,  1.68it/s, loss=16.055, ppl| WARNING: overflow detected, setting loss scale to: 32.0
| epoch 001:   1%| | 22/1864 [00:15<17:56,  1.71it/s, loss=15.796, ppl

**Comment by lematt1991** [comment]
You can use a combination of `--reduce-lr-on-plateau` and `--min-lr` to accomplish this.

**Comment by DonnieZhang586** [comment]
When I use this parameter, it appears error：unrecognized arguments:--reduce-lr-on-plateau

**Comment by lematt1991** [comment]
Sorry, actually it looks like it should be `--lr-scheduler redue_lr_on_plateau`.  That said, I forgot that #904, added `--patience <int>`, which will stop training if the validation loss does not improve after a certain number of epochs.  This may be more fitting for what you are trying to accomplish.  Hope this helps!
